pkg/frontend/adminactions/drain.go

```diff
@@ -0,0 +1,58 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"log"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/drain"
+)
+
+func (k *kubeActions) CordonNode(ctx context.Context, nodeName string, unschedulable bool) error {
+	node, err := k.kubernetescli.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	drainer := &drain.Helper{
+		Ctx:                 ctx,
+		Client:              k.kubernetescli,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Timeout:             60 * time.Second,
+		DeleteEmptyDirData:  true,
+		DisableEviction:     true,
+		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+			log.Printf("deleted pod %s/%s", pod.Namespace, pod.Name)
+		},
+		Out:    log.Writer(),
+		ErrOut: log.Writer(),
+	}
+
+	return drain.RunCordonOrUncordon(drainer, node, unschedulable)
```

Comment:
nit - feel free to ignore & resolve. This appears to be the final resting place of `unschedulable`, it might be clearer to rename this variable everywhere to something like `shouldCordon` considering this is it's only use - I feel this change removes a double-negative from a boolean parameter and better states the purpose without translation.

Reply:
I agree with @sudobrendan here. Since the function is called `CordonNode` than the flipping param would be more understandable if named `shouldCordon`.

Reply:
changed parameter name